### DISCUSSION TITLE
.env file overwritten when saving compose within Dockge

### DISCUSSION THREAD

**Discussion by CollapsingStar** [louislam/dockge#315] [Ask for Help] [answered]
### ⚠️ Please verify that this bug has NOT been raised before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### 📝 Describe your problem

I have an .env file in its corresponding stack folder that I want to edit. It seems that if I stop the stack, edit the .env, and initiate an "Edit" and "Save" in Dockge, the .env file is reverted to a previous state. Steps to reproduce:

- "Stop & Down" or "Stop" the stack within Dockge
- Edit the .env file
- Return to Dockge (with the stack still down), and click "Edit", then immediately click "Save".
- .env is reverted to before changes were made

The overwrite happens exactly when "Save" is selected, not "Edit", which I confirmed by catting the .env after each step. It seems the .env is going back to a previously cached state somehow. As a test, I opened all permissions on the .env and the behavior is still the same. I cannot reproduce this issue with bare Docker CLI commands.

Anyone else experiencing this behavior? Is there a reason a compose file edit within Dockge is changing the .env?

### 📝 Error Message(s) or Log

Stopped/down stack, cat file after editing "this is an edit":

![image](https://github.com/louislam/dockge/assets/91231759/630f0705-97a7-4fe7-b025-f04c7da49099)

Click edit:

![image](https://github.com/louislam/dockge/assets/91231759/768a5563-bada-4db3-95c5-c8a93c89eaa4)

Click save:

![image](https://github.com/louislam/dockge/assets/91231759/49480553-f4f5-4d7b-97fc-14b275ffc5e1)

.env reverted back to previous state:

![image](https://github.com/louislam/dockge/assets/91231759/075944fd-4adb-4e57-9273-1980c26e4a9f)

### 🐻 Dockge Version

1.3.5

### 💻 Operating System and Arch

Ubuntu 22.04

### 🌐 Browser

Firefox

### 🐋 Docker Version

24.0.7

### 🟩 NodeJS Version

N/A

**Comment by louislam** [comment, marked as answer]
You meant you had edited the .env file using another text editor?

If so, you should refresh the page after that, because Dockge do not watch for file changes in real-time currently.

**Reply by CollapsingStar** [reply]
Correct, I edited in another file editor. I didn't realize until this morning that the .env could be edited in the stacks page (due to my compose file being extremely long, I had to scroll down quite a bit), so I'll edit within Dockge from now on. Thanks!
